src/components/AddingToCartOverlay/index.tsx

```diff
@@ -0,0 +1,59 @@
+import { Center, Spinner, useToast, Flex, Text, Box } from "@chakra-ui/react";
+import { useState, useEffect } from "react";
+import { useDispatch } from "react-redux";
+import { switchAddingToCart, switchConstructionMounted } from "@/store/reducer/buttonToggleSlice";
+import { changeConstructionPdfSrc } from "@/store/reducer/constructionSlice";
+
+const AddingToCartOverlay = () => {
+  const dispatch = useDispatch();
+  const [count, setCount] = useState(10);
+  const toast = useToast();
+  if (count <= 0) {
+    toast({
+      title: `Adding to cart failed, please try again`,
+      status: "error",
+      duration: 9000,
+      isClosable: true,
+    });
+  }
+  useEffect(() => {
+    if (count <= 0) {
+      dispatch(switchAddingToCart(false));
+      dispatch(switchConstructionMounted(false));
+      dispatch(changeConstructionPdfSrc(null));
+      return;
+    }
+    const timerId = setInterval(() => {
+      setCount((prevCount) => prevCount - 1);
+    }, 1000);
+    return () => {
+      clearInterval(timerId);
+    };
+  }, [count]);
+  return (
+    <Center
+      position="fixed"
+      h="100vh"
+      w="100vw"
+      bottom="0"
+      left="0"
+      bg="blackAlpha.600"
+      zIndex="5999"
+    >
+      <Flex alignItems="center">
+        <Box width="100px">
+          <Spinner size="xl" color="teal.300" />
+        </Box>
+        <Text fontSize="24" color="teal.300">
+          {count > 8
+            ? "Generating design"
+            : count > 6
+            ? "Generating construction"
+            : "Adding to cart"}
+        </Text>
```

Comment:
So if Im not wrong, this part mimics a countdown process on the screen to show users the progress of the adding cart function. But why doing this? Or is this function mentioned on the Jira board? 
I mean if it is truly necessary to make an overlay, there are components from the chakra UI right? (maybe Modal component?)

Reply:
For the first question, we need to disable all operations that may modify the design when add-to-cart is processing. Any changes made to the design during this period will trigger a new add-to-cart process, which is not what we want. The countdown information is just for making the entire process more user-friendly. 
For the second question, I tried Modal but it seems it will be a little bit jerking due to its rendering machenism. My thought is an overlay also makes sense as we only need an overlay rathan than a modal with other components.